### PR TITLE
Retain file information when closing and reopening

### PR DESCRIPTION
**Pull request by camila314**
Previously, closing and reopening the program meant having to re-save the opened file and overwrite the file. This retains the opened file so you no longer have to go through that process.

Also changed "Clear the watch list" to "Reset watch list" and made it discard the saved file information.

### CODE REVIEW

**Comment by dreamsyntax** [comment]
I like the intent behind this, but I still would prefer something more like this: https://github.com/aldelaro5/dolphin-memory-engine/issues/121
Your change is basically what I want, but I also wanted a way to toggle the modes, incase working in a scratch space.

The clear/reset change sounds good.

**Comment by camila314** [comment]
the best system would be to make it document-based (like Bit Slicer for macOS) but i fear i don't know enough about qt to make that change, maybe some day though

**Comment by dreamsyntax** [comment]
Just to be clear, so we don't break the flow of existing users I would want
https://github.com/aldelaro5/dolphin-memory-engine/issues/121#issuecomment-2106374792

```
File > [x] Autoload Last File
     > [x] Autosave Last File
```

Something like that in the PR that enables this behavior.
It could just be one toggle.
File -> [x] Auto Load/Save File

**Comment by dreamsyntax** [comment]
Bumping.

@cristian64 thoughts on just merging as is? I like the general idea even if I prefer optionally maintaining the old way.

File: Source/GUI/MemWatcher/MemWatchWidget.cpp [hunk 1]
```diff
@@ -267,7 +267,7 @@ void MemWatchWidget::onMemWatchContextMenuRequested(const QPoint& pos)
     contextMenu->addSeparator();
   }
 
-  if (!node || node->isGroup())
+  if (node == m_watchModel->getRootNode() || node->isGroup())
```
Comment (cristian64) [inline Source/GUI/MemWatcher/MemWatchWidget.cpp:270]: _Ideally_, this change (which relates to https://github.com/aldelaro5/dolphin-memory-engine/pull/175) should land in its own PR.

File: Source/GUI/MemWatcher/MemWatchWidget.cpp [hunk 2]
```diff
@@ -853,6 +853,7 @@ void MemWatchWidget::clearWatchList()
     return;
 
   m_watchModel->clearRoot();
+  m_watchListFile = "";
```
Comment (cristian64) [inline Source/GUI/MemWatcher/MemWatchWidget.cpp:856]: To avoid constructing another object for the sake of clearing the member:

```diff
- m_watchListFile = "";
+ m_watchListFile.clear();
```

File: .gitignore [hunk 3]
```diff
@@ -17,4 +17,5 @@ Source/.vs/
 Source/.vscode/
 .vscode/
 .idea/*
-.DS_Store
\ No newline at end of file
+.DS_Store
+.cache
```
Comment (cristian64) [inline .gitignore:21]: Out of curiosity, what is `.cache`?

And, as always, these unrelated changes should at the very least have their own commit and description. Perhaps it can even have a dedicated PR to avoid getting the change blocked by other potentially controversial changes.

**Comment by cristian64** [comment]
> Bumping.
> 
> @cristian64 thoughts on just merging as is? I like the general idea even if I prefer optionally maintaining the old way.

I may be a bit out of the loop, but I'm not entirely sure what these are trying to accomplish. By looking at the diff, I can't see how the watch list file path is consumed on startup.

**Comment by dreamsyntax** [comment]
@camila314 following up if you plan to make changes, I'm going to release another patch version soon
